nakedret.go

```diff
@@ -210,26 +273,52 @@ func (v *returnsVisitor) Visit(node ast.Node) ast.Visitor {
 		funcName = fmt.Sprintf("<func():%v>", file.Position(s.Pos()).Line)
 	case *ast.ReturnStmt:
 		// We've found a possibly naked return statement
-		if v.reportNaked && len(s.Results) == 0 {
-			file := v.f.File(s.Pos())
-			log.Printf("%v:%v: %v naked returns on %v line function\n", file.Name(), file.Position(s.Pos()).Line, v.funcName, v.funcLength)
-			v.root.found = true
+		fun := v.stack[len(v.stack)-1]
+		funName := nestedFuncName(v.stack)
+		if fun.reportNaked && len(s.Results) == 0 && push {
+			sFix := nakedReturnFix(s, fun.funcType)
+			b := &bytes.Buffer{}
+			err := printer.Fprint(b, v.f, sFix)
+			if err != nil {
+				log.Printf("failed to format named return fix: %s", err)
+			}
+			//v.pass.Reportf(s.Pos(), "%v naked returns on %v line function", funName, fun.funcLength)
+			//v.pass.Reportf(s.Pos(), "naked return in func `%s` with %d lines of code", funName, fun.funcLength)
+			v.pass.Report(analysis.Diagnostic{
+				Pos:     s.Pos(),
+				End:     s.End(),
+				Message: fmt.Sprintf("naked return in func `%s` with %d lines of code", funName, fun.funcLength),
+				SuggestedFixes: []analysis.SuggestedFix{{
```

Comment:
SuggestedFixes do not yet get picked up by golangci-lint, but it's planned,
https://github.com/golangci/golangci-lint/issues/1779#issuecomment-1451023670:

> I think that it's the right thing for a linter to support SuggestedFixes (even if SuggestedFixes is still [an experimental API](https://github.com/golang/tools/blob/fbb25cbbd5ff5845639a3035f1fb7ddfd0ac0624/go/analysis/diagnostic.go#L49)), I also believe that we have to remove our custom implementation.